src/generators/schema/author.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package Yoast\WP\Free\Presentations\Generators\Schema
+ */
+
+namespace Yoast\WP\Free\Presentations\Generators\Schema;
+
+use Yoast\WP\Free\Context\Meta_Tags_Context;
+use Yoast\WP\Free\Helpers\Article_Helper;
+use Yoast\WP\Free\Helpers\Image_Helper;
+use Yoast\WP\Free\Helpers\Schema;
+
+/**
+ * Returns schema Person data.
+ *
+ * @since 10.2
+ */
+class Author extends Person {
+	/**
+	 * The Schema type we use for this class.
+	 *
+	 * @var string[]
+	 */
+	protected $type = [ 'Person' ];
+
+	/**
+	 * @var Article_Helper
+	 */
+	private $article_helper;
+
+	/**
+	 * Author constructor.
+	 *
+	 * @param Article_Helper $article_helper
+	 */
+	public function __construct(
+		Article_Helper $article_helper,
+		Image_Helper $image_helper,
+		Schema\Image_Helper $schema_image_helper
+	) {
+		parent::__construct( $image_helper, $schema_image_helper );
+		$this->article_helper = $article_helper;
+	}
+
+	/**
+	 * Determine whether we should return Person schema.
+	 *
+	 * @param Meta_Tags_Context $context The meta tags context.
+	 *
+	 * @return bool
+	 */
+	public function is_needed( Meta_Tags_Context $context ) {
+		if ( $context->indexable->object_type === 'author' ) {
+			return true;
+		}
+
+		if (
+			$context->indexable->object_type === 'post' &&
+			$this->article_helper->is_article_post_type( $context->indexable->object_sub_type )
+		) {
+			// If the author is the user the site represents, no need for an extra author block.
+			if ( (int) $context->post->post_author === $context->site_user_id ) {
```

Comment:
When you change this to a `!==` you can return the value of this check directly.